### PR TITLE
fix(onboarding): define child handoffs and fresh demo runs

### PR DESCRIPTION
**Pull request by davincios**
User-requested onboarding improvement; no existing issue.

#### Describe the changes you have made in this PR -

Onboarding now resolves a demo choice and hands execution to its child, with explicit completion conditions and child-owned live progress. Capability questions receive an answer and an offer to run `/demo`; explicit specialist requests go directly to their skill. An unavailable picker ends guided onboarding without a text-menu replacement, and failed child loading retains the choice without claiming a completed handoff.

The picker offers CI analysis, scheduling, Slack, and Skip. Managed-service availability remains directly accessible as a skill but is no longer a selectable demo. Skill names remain unchanged; Slack's directory and menu order now match its third position.

An explicit `/demo` starts a fresh run. Session question ownership allows the child to ask its questions again while preserving unrelated answered questions. A real-loop regression reproduced the previous failure: the top-level picker reopened, but the child's already-answered repository question was refused. The analytics child's existing scan, repository picker, and five-step plan are unchanged. Session-facet tests now compare the actual inherited field set, and skill metadata accepts the current calendar date in any editor timezone while continuing to reject future dates.

### Demo/Screenshot for feature changes and bug fixes -

Actual menu loaded from the bundled skill:

```text
Which demo would you like me to run?
1. Explore a repo and analyze its CI/CD performance (recommended)
2. Set up an agent that improves CI/CD reliability over time
3. Connect OpenSRE to Slack and hand off DevOps chores for your team
4. Skip the demo and open the shell
```

Offline real-loop coverage verifies that the entry hook pauses execution until selection, the selected child begins work in the answer turn, and a fresh demo can repeat the repository question. Terminal-loss coverage verifies that onboarding exits without rendering a fallback list. These use scripted model responses and real skill loading/hooks; natural-language routing changes were additionally forward-reviewed against the assembled instructions.

Validation:

- `make lint`
- `make format-check`
- `make typecheck` — 1,865 source files
- `uv run python -m pytest core/agent_harness/prompts/skills/ tests/core/agent_harness/prompts/ tests/core/agent/prompts/ tests/core/agent_harness/session/ tests/interactive_shell/runtime/test_demo_picker.py tests/interactive_shell/command_registry/test_choice_prompt_skip.py tests/tools/interactive_shell/test_skill_entry.py tests/packaging/test_release_manifest.py -q` — 239 passed, 1 existing xfailed
- `uv run python -m pytest tests/core/agent/orchestration/test_ask_choice_tool.py tests/core/agent/orchestration/test_menu_turn_end.py tests/interactive_shell/command_registry/test_choice_prompt_demo_repository.py tests/interactive_shell/command_registry/test_slash_catalog.py tests/shared/test_surface_border.py tests/shared/test_integrations_api_border.py tests/shared/test_tool_api_border.py -q` — 59 passed

- `TZ=UTC uv run python -m pytest tests/interactive_shell/runtime/test_demo_repository.py tests/interactive_shell/session/test_session_facets.py tests/core/agent_harness/prompts/test_skill_metadata.py tests/core/agent/orchestration/test_action_prompt.py -q` — 49 passed; reproduces and covers the first CI run's failures

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

Codex implemented and self-reviewed this change after a user-approved design interview. Human review remains pending. The router owns selection and skill loading, while each child retains its execution, prerequisites, and authorization. Shared prompt rules agree with those boundaries. Runtime changes are limited to tracking queued question ownership for fresh demo runs and ending onboarding if its terminal becomes unavailable; existing plan-completion rules are preserved.

---

## Checklist before requesting a review
- [x] I have added a proper PR title (user-requested work; no existing issue)
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by davincios** [comment]
@greptile review

**Comment by davincios** [comment]
@codex review

File: core/agent_harness/prompts/skills/onboarding-github-ci/SKILL.md [hunk 1]
```diff
@@ -1,87 +1,114 @@
...
+  through a demo picker to the selected child skill. Answer capability
+  questions first and offer /demo. Load the specialist directly when the
+  user names a demo or requests repository analysis, recurring CI fixes,
+  Slack setup, managed-service availability, or a one-off CI fix.
 metadata:
   owner: Vincent
-  last_changed_by: Vincent
-  last_changed_at: 2026-09-11
+  last_changed_by: Jan
+  last_changed_at: 2026-09-12
```
[the remark this answers is not in the document]
Comment (davincios) [reply core/agent_harness/prompts/skills/onboarding-github-ci/SKILL.md:12]: Addressed in 60a5dd21c. The edit was made on 2026-09-12 in Europe/Berlin (UTC+02); the CI/review timestamp is 2026-09-11 in UTC. The cards contain the actual local calendar date required by the skill instructions.

The failure was the validator using the runner's calendar date. It now compares against the latest current calendar date (UTC+14), accepting an editor's valid date while continuing to reject dates that are future in every timezone. The exact failures reproduced locally with `TZ=UTC`; all 49 tests in the affected files now pass with that setting. CI Gate and all test shards pass on 60a5dd21c: https://github.com/Tracer-Cloud/opensre/actions/runs/34655117068.

**Comment by davincios** [comment]
@greptile review
